unit_tests/dummy_remote.py

```diff
@@ -41,6 +47,58 @@ def receive_files(src, dst):
         return True
 
 
+class DummyRemoteWithException(DummyRemote):
+
+    def __init__(self) -> None:
+        # Hold exception object that should be raised
+        self._test_exception: Exception or None = None
+        # How many time return exception from run
+        self._exception_amount: int = 0
+        # How many times the "run" function has been called.
+        self._run_number: int = 0
+
+    def run(self, *args, **kwargs):
+        logging.info(args, kwargs)
+        self.raise_exception()
+
+        if 'sudo coredumpctl --no-pager' in args[0]:
+            dummy_output = "No coredumps found"
+        else:
+            dummy_output = args[0]
+        return DummyOutput(dummy_output)
+
+    def raise_exception(self):
+        if self.test_exception:
+            while self.exceptions_amount:
```

Comment:
It's effectively `if`

Reply:
The function raise_exception is not called "exceptions_amount" times. It raises needed amount of exceptions and exit. So it should be loop

Reply:
You can't raise exceptions in a loop :) The first one will break it.

Reply:
you are right. fixed